emoji_picker: Fix status emoji picker inconsistencies.

sayamsamal opened the pull request:
Due to a logical bug in the `process_enter_while_filtering` function, the `toggle_reaction` was being called when pressing enter while filtering the emojis in the status emoji picker.

This PR fixes the above bug, while also adding some cleanups to the `emoji_picker.js` by combining multiple click handlers and aggregating all the logic related to an emoji being selected into a common function.

Fixes: #28464

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

**Steps to reproduce**: Go to "Set  Modal > Open emoji picker > Enter random words to filter > Press Enter

[before-status-emoji-picker.webm](https://github.com/zulip/zulip/assets/82862779/bb35aae7-0ab7-4daf-9ab5-8f4e4a760d57)

[after-status-emoji-picker.webm](https://github.com/zulip/zulip/assets/82862779/d83652c0-760b-4511-8759-28b691a9b46a)

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [X] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [X] Explains differences from previous plans (e.g., issue description).
- [X] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [X] Each commit is a coherent idea.
- [X] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [X] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [X] End-to-end functionality of buttons, interactions and flows.
- [X] Corner cases, error conditions, and easily imagined bugs.
</details>

timabbott commented on web/src/emoji_picker.js:
```diff
@@ -699,34 +666,72 @@ export function toggle_emoji_popover(target, id, additional_popover_options) {
+
+function handle_emoji_clicked($emoji, event) {
+    const emoji_name = $emoji.attr("data-emoji-name");
+    const emoji_type = $emoji.attr("data-emoji-type");
```
This looks great except that we really don't need a `data-emoji-type` on every single emoji element in the picker, and that name is confusing given it is not equal to `emoji.type`.

I'm pretty sure we don't allow multiple emoji picker popovers open at the same time, so simplest would just be to store this value in a global variable or an attribute on the popover as a whole. As a global variable, I'd call it something like `emoji_destination`.

sayamsamal replied:
Renamed the data attribute to `data-emoji-destination` and shifted it to `.emoji-picker-popover` div as a global variable.

timabbott commented:
lgtm aside from the small comment above.

sayamsamal commented:
In addition to the addressing the feedback, I have also added another commit which aims to restructuring of related CSS and removing the class names -- "reaction", "status-emoji" and "composition" from the `.emoji-popover-emoji` div, which were previously being used for handling click events.

timabbott commented on web/styles/dark_theme.css:
```diff
@@ -515,17 +515,34 @@
+            background-color: hsl(0deg 0% 0% / 50%);
+            border-color: hsl(0deg 0% 0% / 90%);
+        }
+    }
```
This isn't new with this PR, but this is the focus+reacted behavior that I see in the emoji picker for reactions:

![image](https://github.com/zulip/zulip/assets/2746074/b6a364d6-2099-4685-9223-64c4ad57f976)

That's probably a bug we should fix in a follow-up?

timabbott commented:
Merging since these changes are themselves good, and the one bug I found is present in `main`. Can you put up a fix for it?
